coderd/provisionerdaemons.go

```diff
@@ -249,6 +253,46 @@ func (server *provisionerdServer) AcquireJob(ctx context.Context, _ *proto.Empty
 				},
 			},
 		}
+	case database.ProvisionerJobTypeTemplateVersionPlan:
+		var input templateVersionPlanJob
+		err = json.Unmarshal(job.Input, &input)
+		if err != nil {
+			return nil, failJob(fmt.Sprintf("unmarshal job input %q: %s", job.Input, err))
+		}
```

Comment:
It scares me to add another job with such similar scope to the others, but I understand why.

It'd be helpful if you explained why we couldn't use the `workspace_build` job to do similarly. I believe it'd just require not being attached to a literal workspace build, which seems like a reasonable change anyways.

Reply:
We need to have a separate job for this because the job endpoints have different schemas and different endpoints to access the job details, not a unified `/jobs` endpoint. You've said in the past that you'd like to split these different job types apart more in the future (so they don't share DB tables for metadata etc.), so merging these two jobs together would be counterintuitive to that goal

Reply:
The inputs are slightly different, but the outputs are the same, right?